File: C++/word_counter/ver_python/topk.py
# topk.py
import time
from collections import Counter

def main():
    start = time.perf_counter()

    with open("../words_data/words.txt") as f:
        counter = Counter(f.read().splitlines())

    top10 = counter.most_common(10)

    elapsed = time.perf_counter() - start
    print("Top10:", top10)
    print("Time:", elapsed)

if __name__ == "__main__":
    main()

File: C++/word_counter/ver_C++/topk.cpp
// topk.cpp
#include <iostream>
#include <fstream>
#include <unordered_map>
#include <vector>
#include <algorithm>
#include <chrono>

int main() {
    using namespace std;
    auto start = chrono::high_resolution_clock::now();

    ifstream file("../words_data/words.txt");
    unordered_map<string, size_t> counts;
    string word;

    while (getline(file, word)) {
        counts[word]++;
    }

    vector<pair<string, size_t>> vec(counts.begin(), counts.end());

    partial_sort(vec.begin(), vec.begin() + 10, vec.end(),
        [](const auto& a, const auto& b) {
            return a.second > b.second;
        });
    
        auto end = chrono::high_resolution_clock::now();
        chrono::duration<double> elapsed = end - start;

        cout << "Top10:\n";
        for (int i = 0; i < 10; ++i) {
            cout << vec[i].first << " : " << vec[i].second << "\n";
        }

        cout << "Time: " << elapsed.count() << " sec\n";
}